src/graph_notebook/notebooks/04-Machine-Learning/Neptune-ML-SPARQL/neptune_ml_sparql_utils.py

```diff
@@ -0,0 +1,701 @@
+
+        genre_df = pd.DataFrame(genres, columns=['~id'])
+        genre_df['~label'] = 'genre'
+        genre_df['name'] = genre_df['~id']
+
+        movie_rdf_filename = os.path.join(self.formatted_directory, 'movies.nq')
+        movie_genre_rdf_filename = os.path.join(self.formatted_directory, 'movie_genres.nq')
+        movie_graph = ConjunctiveGraph()
+        movie_genre_graph = ConjunctiveGraph()
+
+        # movie vertex file creation
+        for index, row in movies_df.iterrows():
+            id = row['~id']
+            title = row['title']
+            imdb_url = row['imdb_url']
+            release_date = row['release_date']
+
+            movie_graph.add((
+                self.ns_resource[id], RDF.type, self.ns_ontology.Movie, self.ns_ontology.Movies
+            ))
+            movie_graph.add((
+                self.ns_resource[id], self.ns_ontology.title, Literal(title, datatype=XSD.string),
+                self.ns_ontology.Movies
+            ))
+            movie_graph.add((
+                self.ns_resource[id], RDFS.label, Literal(title, datatype=XSD.string), self.ns_ontology.Movies
+            ))
+            movie_graph.add((
+                self.ns_resource[id], self.ns_ontology.imdbURL, Literal(imdb_url, datatype=XSD.anyURI),
+                self.ns_ontology.Movies
+            ))
+            movie_graph.add((
+                self.ns_resource[id], self.ns_ontology.releaseDate,
+                Literal(release_date, datatype=XSD.dateTime), self.ns_ontology.Movies
+            ))
+            # add genre labels
+            for genre_value in genres:
+                if (id != "movie_172") and (id != "movie_235") and (id != "movie_121"):
+                    if row[genre_value]:
+                        movie_genre_graph.add((
+                            self.ns_resource[id], self.ns_ontology.hasGenre,
+                            Literal(genre_value, datatype=XSD.string), self.ns_ontology.Movie
+                        ))
+                elif row[genre_value]:
+                    movie_genre_graph.add((
+                        self.ns_resource[id], self.ns_ontology.hasOriginalGenre,
+                        Literal(genre_value, datatype=XSD.string), self.ns_ontology.Movie
+                    ))
+
+        movie_graph.serialize(format='nquads', destination=movie_rdf_filename)
+        movie_genre_graph.serialize(format='nquads', destination=movie_genre_rdf_filename)
+
+    def __process_users_rdf(self):
+        print("Processing Users to RDF")
+        # User Vertices - Load, rename column with type, and save
+
+        user_df = pd.read_csv(os.path.join(
+            self.raw_directory, 'ml-100k/u.user'), sep='|', encoding='ISO-8859-1',
+            names=['~id', 'age:Int', 'gender', 'occupation', 'zip_code'])
+        user_df['~id'] = user_df['~id'].apply(
+            lambda x: f'user_{x}'
+        )
+
+        users_graph = ConjunctiveGraph()
+
+        for index, row in user_df.iterrows():
```

Comment:
Same comment as above on `iterrows()`

Reply:
Removing this line resulted in an error, so i have replaced it.